Fix implementation of wxScrolled<wxControl>

a-wi opened the pull request:
wxControl::Create has a different signature than wxWindow::Create
(its 6-th parameter is of const wxValidator& type instead of
const wxString&) so it cannot be invoked from the the general template of
wxScrolled<T>::Create method. We need to have a template specialization of
this method for wxControl from where wxControl::Create could be called
with rearranged parameters.

vadz commented:
I don't like duplicating `Create()`, even if it's relatively short, so let me propose 2 alternative fixes:

1. Very special one: just add overloaded `Create()` taking 6 parameters to `wxControl`. If it doesn't use any default values for its arguments, there shouldn't be any ambiguities/compatibility problems AFAICS.
2. More general one: currently `wxScrolled<T>` requires `T` to have the ctor with this special signature, which may also prevent it from being used with custom classes. We could add `wxCreateScrolled()` template function that would just call `T::Create()` by default but could be overloaded for any custom type -- and provide its overload for `wxControl`. This would avoid duplicating the rest of `Create()` and be more flexible.

What do you think?

a-wi commented:
Indeed, option 2 seems to be the most elegant solution so I've tried to implement it.

vadz commented:
I was actually thinking about making `wxCreateScrolled()` a (global) function in order to allow overloading (instead of specializing) it, but I don't really know if this has any important advantages... However if we keep it as a member, we should rename it to `CreateScrolledBase()` to be consistent with the member functions naming convention.

It would also be better to add a comment before the specialization (or overload if you decide to change this) to explain why do we need it.

a-wi commented:
Using a global function would be indeed the most clear implementation but how to invoke non-static member function T::Create() from global wxCreateScrolled()?
Is it any trick to avoid "cannot call member function without object" error with this code:
`template<class T>`
`bool wxCreateScrolled(params)`
`{`
`    return T::Create(params);`
`}`

vadz commented:
It just needs to take `T* self` as parameter and call `self->Create(rest-of-params)`. And then it would be called as `wxCreateScrolled(this, ...)`.

a-wi committed:
Fix implementation of wxScrolled<wxControl>

wxControl::Create() has a different signature than wxWindow::Create()
(its 6-th parameter is of const wxValidator& type instead of
const wxString&) so it cannot be invoked from the the general template of
wxScrolled<T>::Create() method. We need to move a call to T::Create()
function to a dedicated template function wxCreateScrolled() responsible
for actual creation of the scrolled window and overload it for custom
classes if necessary.
This has to be done for wxControl and from this overloaded function
wxControl::Create() can be called with rearranged parameters.

a-wi commented:
It seems I'm done with implementation. I also updated wxScrolled documentation - please check if what I wrote is fine.
